src/NetTopologySuite/Geometries/CoordinateSequences.cs

```diff
@@ -69,14 +240,22 @@ public static void Copy(CoordinateSequence src, int srcPos, CoordinateSequence d
         /// <param name="destPos">The index of the coordinate in <see paramref="dest"/></param>
         public static void CopyCoord(CoordinateSequence src, int srcPos, CoordinateSequence dest, int destPos)
         {
-            int minDim = Math.Min(src.Dimension, dest.Dimension);
-            for (int dim = 0; dim < minDim; dim++)
+            GetCommonOrdinateIndices(src, dest, out int[] srcIndices, out int[] destIndices);
+            CopyCoord(src, srcPos, srcIndices, dest, destPos, destIndices);
+        }
+
+        private static void CopyCoord(CoordinateSequence src, int srcPos, int[] srcIndices,
+                                      CoordinateSequence dest, int destPos, int[] destIndices)
+        {
+            if (srcIndices.Length != destIndices.Length)
+                throw new ArgumentException(nameof(destIndices));
```

Comment:
(nit) this constructor doesn't set `ParamName`, but rather it sets the `Message`.